src/server/status/metrics_collector/metrics_collector.js

```diff
@@ -1,113 +0,0 @@
-/*
- * Licensed to Elasticsearch B.V. under one or more contributor
- * license agreements. See the NOTICE file distributed with
- * this work for additional information regarding copyright
- * ownership. Elasticsearch B.V. licenses this file to you under
- * the Apache License, Version 2.0 (the "License"); you may
- * not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-import { Metrics } from './metrics';
-
-const matchSnapshot = /-SNAPSHOT$/;
-
-/*
- * Persist operational data for machine reading
- * sets the latest gauge values
- * sums the latest accumulative values
- */
-export class MetricsCollector {
-  constructor(server, config) {
-
-    // NOTE we need access to config every time this is used because uuid is managed by the kibana core_plugin, which is initialized AFTER kbn_server
-    this._getBaseStats = () => ({
-      name: config.get('server.name'),
-      uuid: config.get('server.uuid'),
-      version: {
-        number: config.get('pkg.version').replace(matchSnapshot, ''),
-        build_hash: config.get('pkg.buildSha'),
-        build_number: config.get('pkg.buildNum'),
-        build_snapshot: matchSnapshot.test(config.get('pkg.version'))
-      }
-    });
-
-    this._stats = Metrics.getStubMetrics();
-    this._metrics = new Metrics(config, server); // TODO: deprecate status API that uses Metrics class, move it this module, fix the order of its constructor params
-  }
-
-  /*
-   * Accumulate metrics by summing values in an accumulutor object with the next values
-   *
-   * @param {String} property The property of the objects to roll up
-   * @param {Object} accum The accumulator object
-   * @param {Object} next The object containing next values
-   */
-  static sumAccumulate(property, accum, next) {
```

Comment:
For background on why there is so much deleted code:

In this phase of providing a stats API, we want something that is in parity with the data model and the **meaning** of the data that is currently indexed into the `.monitoring-kibana-*` indices. This PR is a step towards being able to externalize the collection of the stats. This deleted code was not meeting the goal of achieving parity because it provided stats that constantly accumulate. That is, the number of requests, max response time, are kept in memory and new data from ops events add to it. In the long term, that is what it should do as it provides the best way to store machine-readable metrics.

This PR removes the `MetricsCollector` class with the accumulation metrics, and moves to providing the **latest** data that has been received from the ops events. Therefore, the meaning of the data stays the same as what is currently indexed in `.monitoring-kibana-*`.